src/GenSymIO.chpl

```diff
@@ -466,7 +466,22 @@ module GenSymIO {
                     }
                     read_files_into_distributed_array(entryInt.a, subdoms, filenames, dsetName);
                     var rname = st.nextName();
-                    st.addEntry(rname, entryInt);
+                    
+                    /*
+                     * Since boolean pdarrays are saved to and read from HDF5 as ints, confirm whether this
+                     * is actually a boolean dataset. If so, (1) convert the SymEntry pdarray to a boolean 
+                     * pdarray, (2) create a new SymEntry of type bool, (3) set the SymEntry pdarray 
+                     * reference to the bool pdarray, and (4) add the entry to the SymTable
+                     */
+                    if isBooleanDataset(filenames[0],dsetName) {
+                        var a_bool = entryInt.a:bool;
+                        var entryBool = new shared SymEntry(len, bool);
+                        entryBool.a = a_bool;
```

Comment:
You can get rid of a copy here by deleting line 477 and changing 479 to:

```chpl
entryBool.a = entryInt.a:bool;
```

Reply:
Good point, fixed and pushed to git